lib/bokken_web/auth/bot_manager.ex

```diff
@@ -0,0 +1,30 @@
+defmodule BokkenWeb.Auth.BotManager do
+  use Guardian, otp_app: :bokken
+
+  alias Bokken.Accounts
+
+  def subject_for_token(bot, _claims) do
+    {:ok, to_string(bot.id)}
+  end
+
+  def resource_from_claims(%{"sub" => id, "exp" => exp}) do
+    with {:ok, bot} <- Accounts.get_bot(id),
+         {:ok, _} <- verify_exp(exp) do
+      {:ok, bot}
+    else
+      _ -> {:error, :unauthorized}
+    end
+  rescue
+    Ecto.NoResultsError -> {:error, :resource_not_found}
+  end
+
+  defp verify_exp(exp) do
```

Comment:
I think you don't need to do so much work by your self. Guardian has the expiration time, algs etc already built and well tested. 

It should be something like `ttl: {90, :day}` :thinking: 

https://hexdocs.pm/guardian/Guardian.Token.Jwt.html

Reply:
https://github.com/ueberauth/guardian#optional-configuration-jwt

All configuration including ttl is available for the JWT type :point_up:

Reply:
You're right. I'll take a look into it!